syncer/checkpoint.go

```diff
@@ -439,14 +490,22 @@ func (cp *RemoteCheckPoint) CheckGlobalPoint() bool {
 }
 
 // Rollback implements CheckPoint.Rollback
-func (cp *RemoteCheckPoint) Rollback() {
+func (cp *RemoteCheckPoint) Rollback(schemaTracker *schema.Tracker) {
 	cp.RLock()
 	defer cp.RUnlock()
 	cp.globalPoint.rollback()
 	for schema, mSchema := range cp.points {
 		for table, point := range mSchema {
 			cp.logCtx.L().Info("rollback checkpoint", log.WrapStringerField("checkpoint", point), zap.String("schema", schema), zap.String("table", table))
-			point.rollback()
+			if point.rollback() {
+				// schema changed
+				_ = schemaTracker.DropTable(schema, table)
+				if point.ti != nil {
+					if err := schemaTracker.CreateTableIfNotExists(schema, table, point.ti); err != nil {
+						cp.logCtx.L().Warn("failed to rollback schema on schema tracker", zap.String("schema", schema), zap.String("table", table), log.ShortError(err))
```

Comment:
How can we recover from this situation?

Reply:
Don't know. I don't think there's any normal situation errors would arise. Upgraded the log level to Error so we can catch it if it does happen.